File: jediterm-core-mpp/src/jvmMain/java/com/jediterm/terminal/TtyConnector.java
package com.jediterm.terminal;

import com.jediterm.core.util.TermSize;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public interface TtyConnector {
  int read(char[] buf, int offset, int length) throws IOException;

  void write(byte[] bytes) throws IOException;

  void write(String string) throws IOException;

  boolean isConnected();

  default void resize(@NotNull TermSize termSize) {
    // support old implementations not overriding this method
    resize(new java.awt.Dimension(termSize.getColumns(), termSize.getRows()));
  }

  int waitFor() throws InterruptedException;

  boolean ready() throws IOException;

  String getName();

  void close();

  /**
   * @deprecated use {@link #resize(TermSize)} instead
   */
  @Deprecated
  default void resize(@NotNull java.awt.Dimension termWinSize) {
    // support old implementations overriding neither `resize(Dimension)` nor this method
    resize(termWinSize, new java.awt.Dimension(0, 0));
  }

  /**
   * @deprecated use {@link #resize(TermSize)} instead
   */
  @SuppressWarnings("unused")
  @Deprecated
  default void resize(java.awt.Dimension termWinSize, java.awt.Dimension pixelSize) {
    throw new IllegalStateException("This method shouldn't be called. " +
      getClass() + " should override TtyConnector.resize(com.jediterm.core.util.TermSize)");
  }

  /**
   * @deprecated Collect extra information when creating {@link TtyConnector}
   */
  @SuppressWarnings("removal")
  @Deprecated(forRemoval = true)
  default boolean init(Questioner q) {
    return true;
  }
}

File: jediterm-core-mpp/src/jvmMain/kotlin/com/jediterm/terminal/TtyConnector.kt
package com.jediterm.terminal

import com.jediterm.core.util.TermSize
import java.awt.Dimension
import java.io.IOException

interface TtyConnector {
    @Throws(IOException::class)
    fun read(buf: CharArray?, offset: Int, length: Int): Int

    @Throws(IOException::class)
    fun write(bytes: ByteArray?)

    @Throws(IOException::class)
    fun write(string: String?)

    val isConnected: Boolean

    fun resize(termSize: TermSize) {
        // support old implementations not overriding this method
        resize(Dimension(termSize.columns, termSize.rows))
    }

    @Throws(InterruptedException::class)
    fun waitFor(): Int

    @Throws(IOException::class)
    fun ready(): Boolean

    val name: String?

    fun close()

    @Deprecated("use {@link #resize(TermSize)} instead")
    fun resize(termWinSize: Dimension) {
        // support old implementations overriding neither `resize(Dimension)` nor this method
        resize(termWinSize, Dimension(0, 0))
    }

    @Suppress("unused")
    @Deprecated("use {@link #resize(TermSize)} instead")
    fun resize(termWinSize: Dimension?, pixelSize: Dimension?) {
        throw IllegalStateException(
            "This method shouldn't be called. " +
                    javaClass + " should override TtyConnector.resize(com.jediterm.core.util.TermSize)"
        )
    }

    @Deprecated("Collect extra information when creating {@link TtyConnector}")
    fun init(q: Questioner?): Boolean {
        return true
    }
}
